### PR TITLE
Create Combat System.md

### PR DESCRIPTION
**Pull request by SketchFoxsky**
First Commit to the Combat System will need revision after Damage, Game Instance Controller, Object Health, and Gun Controller is commited.

### CODE REVIEW

File: docs/cck/components/Combat System.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
++ **Health Regeneration Cap:** The amount of Health that regeneration will stop.
+
+##### Armor
++ **Armor Base Amount:** This is the default Armor each player starts with at the begining of a Game or Round.
++ **Armor Max Amount:** This is the Maximum amount of Armor a player can have at any given time.
+
+##### Armor Regeneration
++ **Armor Regeneration Delay:** The time in seconds before the player starts regenerating Armor.
++ **Armor Regeneration Rate:** How much Armor the player receives per second.
++ **Armor Regeneration Cap:** The amount of Armor that regeneration will stop.
```
Comment (BenjaminZehowlt) [inline docs/cck/components/Combat System.md:24]: This is worded strangely. Would "The maximum amount of armor that is regenerated." work better here?

File: docs/cck/components/Combat System.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+##### Reference ID
+
+##### Health
++ **Health Base Amount:** This is the default Health each player starts with at the begining of a Game or Round.
++ **Health Max Amount:** This is the Maximum amount of Health a player can have at any given time.
+
+##### Health Regeneration
++ **Health Regeneration Delay:** The time in seconds before the player starts regenerating Health.
++ **Health Regeneration Rate:** How much Health the player receives per second.
```
Comment (BenjaminZehowlt) [inline docs/cck/components/Combat System.md:14]: This is worded strangely. Would "The amount of health the player recovers per second." work better here?

File: docs/cck/components/Combat System.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
++ **Shield Base Amount:** This is the default Armor each player starts with at the begining of a Game or Round.
++ **Shield Max Amount:** This is the Maximum amount of Armor a player can have at any given time.
+
+##### Shield Regeneration
++ **Shield Regeneration Delay:** The time in seconds before the player starts regenerating Shield.
++ **Shield Regeneration Rate:** How much Shield the player receives per second.
++ **Shield Regeneration Cap:** The amount of Shield that regeneration will stop.
+
+##### General Settings
++ **Friendly Fire:** Enable or Disable the ability for players to attack their own teamates.
```
Comment (BenjaminZehowlt) [inline docs/cck/components/Combat System.md:36]: teamates -> teammates

**Review by BenjaminZehowlt — CHANGES_REQUESTED** [review]

File: docs/cck/components/Combat System.md (hunk 1 above)
Comment (SketchFoxsky) [reply docs/cck/components/Combat System.md:24]: The armor cap is based on the amount the player has, if uncapped it'll keep giving armor to the player until the maximum armor is reached, similar to health and shield.

File: docs/cck/components/Combat System.md (hunk 2 above)
Comment (SketchFoxsky) [reply docs/cck/components/Combat System.md:14]: will reword this entry with your changes.

File: docs/cck/components/Combat System.md (hunk 3 above)
Comment (SketchFoxsky) [reply docs/cck/components/Combat System.md:36]: corrected

**Pull request merged**